include/motor-control/core/motion_controller.hpp

```diff
@@ -34,12 +47,12 @@ class MotionController {
     }
 
     void set_acceleration(uint32_t a) { acc = a; }
-    void set_distance();
+    void set_distance(uint32_t d) { dist = d; }
 
-    void move() {
+    void move(const Move& msg) {
         set_pin(hardware_config.enable);
         set_pin(hardware_config.direction);
-        timer_interrupt_start();
+        queue.try_write(msg);
```

Comment:
this is configured to drop the message if it fails. it's worth considering that this is in a freertos task and it therefore may in fact be fine to use a direct `write` that blocks until the message is sent, because that way we won't lose a message (or might lose it farther back where it's easier to send the error).

Reply:
I see what you're saying. I think I would rather consider this in a follow-up PR because we aren't even really taking into account any overflow in the queue at the moment. Will definitely keep in mind.